graspologic/plot/plot.py

```diff
@@ -671,6 +676,226 @@ def pairplot(
+    title : string, no default
+    context :  None, or one of {talk (default), paper, notebook, poster}
+        Seaborn plotting context
+    font_scale : float, optional, default: 1
+        Separate scaling factor to independently scale the size of the font
+        elements.
+    alpha : float, optional, default: 0.7
+        Opacity value of plotter markers between 0 and 1
+    figsize : tuple
+        the size of the 2d subplots configuration
+    kde : boolean
+        allows for access of KDE argument in histplot
+    """
+    # Handle X and labels
+    if labels is not None:
+        check_X_y(X, labels)
+    # Handle gmm
+    if gmm is None:
+        msg = "You must input a sklearn.mixture.GaussianMixture"
+        raise NameError(msg)
+    Y_, means, covariances = gmm.predict(X), gmm.means_, gmm.covariances_
+    data = pd.DataFrame(data=X)
+    n_components = gmm.n_components
+    # setting up the data DataFrame
+    if labels is None:
+        lab_names = [i for i in range(n_components)]
+        data["labels"] = np.asarray([lab_names[Y_[i]] for i in range(Y_.shape[0])])
+    else:
+        data["labels"] = labels
+    data["clusters"] = Y_
+    # creating a cluster_palette from a stirng  only if labels is given
+    if isinstance(cluster_palette, str) and labels is not None:
+        colors = sns.color_palette(cluster_palette, n_components)
+        cluster_palette = dict(zip(np.unique(Y_), colors))
+    # creating a label_palette from a stirng  only if labels is given
+    # note that these can be same string and mapping would be same for both
+    if isinstance(label_palette, str) and labels is not None:
+        colors = sns.color_palette(label_palette, n_components)
+        label_palette = dict(zip(np.unique(np.asarray(labels)), colors))
+    # must give labels AND palettes
+    if (labels is not None) and (cluster_palette is None or label_palette is None):
+        msg = "if give labels array must give palette dicts or string desciptors"
+        raise ValueError(msg)
+    elif labels is None and (cluster_palette is not None or label_palette is not None):
+        msg = "must give labels if using palettes for labels and clusters"
+        raise ValueError(msg)
+    # This takes care of default case, when no palettes or labels given
+    elif labels is None and cluster_palette is None and label_palette is None:
+        colors = sns.color_palette("Set1", n_components)
+        cluster_palette = dict(zip(np.unique(lab_names), colors))
+        label_palette = dict(zip(np.unique(lab_names), colors))
+
+    with sns.plotting_context(context=context, font_scale=font_scale):
+        dimensions = X.shape[1]
+        # we only want 1 scatter plot for 2 features
+        if X.shape[1] == 2:
+            dimensions = 1
+        pairplot, axes = plt.subplots(
+            dimensions, dimensions, figsize=figsize, squeeze=False
+        )
+        # this will allow for uniform iteration whether axes was 2d or 1d
```

Comment:
if squeeze is false then will you not get a (1,1) array back?

Reply:
yes!!